### ISSUE TITLE
Timeout for cache is too strict, it is always failing for me

### ISSUE THREAD

**Issue by sobolevn** [codingjoe/django-health-check#274]
Currently I am facing an issue with new versions of `django-health-check`.

These lines: 
```python
cache.set('djangohealtcheck_test', 'itworks', 1)
if not cache.get("djangohealtcheck_test") == "itworks":
    raise ServiceUnavailable("Cache key does not match")
```

do raise.
Here: https://github.com/KristianOellegaard/django-health-check/blob/master/health_check/cache/backends.py#L21

Build: https://github.com/wemake-services/wemake-django-template/pull/1332/checks?check_run_id=1241149628

### Using `None` as a timeout and the deleting the key

```python
cache.set('djangohealtcheck_test', 'itworks', None)
if not cache.get("djangohealtcheck_test") == "itworks":
    raise ServiceUnavailable("Cache key does not match")
else:
    cache.delete("djangohealtcheck_test")
```

### Using default timeout

```python
cache.set('djangohealtcheck_test', 'itworks')
if not cache.get("djangohealtcheck_test") == "itworks":
    raise ServiceUnavailable("Cache key does not match")
```

I think this is the best one.

**Comment by codingjoe** [comment]
Hi @sobolevn, I agree. The default timeout might make more sense. Since we use the same cache key, I also don't see any chance the cache could overflow, even with a default timeout of `None` and no cache eviction policy on the cache server.
Could you care to write a patch yourself?
Best, Joe

**Comment by sobolevn** [comment]
> Could you care to write a patch yourself?

Yes, sure.

**Comment by codingjoe** [comment]
Released in [3.14.3](https://pypi.org/project/django-health-check/3.14.3/)

**Comment by sobolevn** [comment]
It is still failing for me 😢 

https://github.com/wemake-services/wemake-django-template/pull/1341

Local testing was successful, but the whole integration does not work. Any ideas what to do?

**Comment by codingjoe** [comment]
Not really, we could add a setting, so you can define a long timeout. But before we do that, are you sure your cache works correctly? Can you jump into a server shell and set and get something from the cache for me? Best, Joe

**Comment by sobolevn** [comment]
Yes, `cache` and `caches['default']` work correctly, they are simply `django.core.cache.backends.locmem.LocMemCache`, so they always work.
Link to settings: https://github.com/wemake-services/wemake-django-template/blob/master/%7B%7Bcookiecutter.project_name%7D%7D/server/settings/components/caches.py

Proof:
<img width="752" alt="Снимок экрана 2020-11-04 в 13 48 49" src="https://user-images.githubusercontent.com/4660275/98102345-98aee280-1ea4-11eb-8880-df77abfd471c.png">

**Comment by codingjoe** [comment]
Wait, `LocMemCache` that doesn't sound right for a production setup. What's your WSGI server? `gunicorn`? If so, the setup won't work. `LocMemCache` means your cache is stored in a single thread and thanks to the GIL also consistent within the same processes. But that's it. If you have multiple processes (as you would with gunicorn) or multiple app servers, your cache isn't consistent (shared) across processes.

Anyhow, that should not affect the test that is failing in your CI suite. Are you sure the cache is set up correctly in the CI suite? Maybe add in `isinstance` assertion to the test, to validate everything is correct up to this point.

**Comment by sobolevn** [comment]
> Wait, LocMemCache that doesn't sound right for a production setup

Yes, but I am talking about dev setup for now. 

> Maybe add in isinstance assertion to the test, to validate everything is correct up to this point.

Done! Here's the test I [wrote](https://github.com/wemake-services/wemake-django-template/blob/master/%7B%7Bcookiecutter.project_name%7D%7D/tests/test_server/test_urls.py#L17):

```python
from django.core.cache import caches
from django.core.cache.backends.locmem import LocMemCache

@pytest.mark.django_db
def test_health_check(client: Client) -> None:
    """This test ensures that health check is accessible."""
    response = client.get('/health/')

    assert isinstance(caches['default'], LocMemCache)
    assert response.status_code == 200
```

It passes with `django-health-check = "3.12.1"`:

```
collected 7 items                                                                    

tests/test_server/test_urls.py .......                                         [100%]

```

Now, let's update `django-health-check` to `3.16.0`. And rerun the tests:

```
collected 7 items                                                                    

tests/test_server/test_urls.py ......F                                         [100%]

====================================== FAILURES ======================================
_________________________________ test_health_check __________________________________
tests/test_server/test_urls.py:20: in test_health_check
    assert response.status_code == 200
E   assert 500 == 200
E    +  where 500 = <TemplateResponse status_code=500, "text/html; charset=utf-8">.status_code
-------------------------------- Captured stderr call --------------------------------
timestamp=None level=None event='Internal Server Error: /health/' logger=None
--------------------------------- Captured log call ----------------------------------
ERROR    health-check:backends.py:49 unavailable: Cache key does not match
Traceback (most recent call last):
  File "/Users/sobolev/Desktop/test_project/.venv/lib/python3.8/site-packages/health_check/backends.py", line 30, in run_check
    self.check_status()
  File "/Users/sobolev/Desktop/test_project/.venv/lib/python3.8/site-packages/health_check/cache/backends.py", line 23, in check_status
    raise ServiceUnavailable("Cache key does not match")
health_check.exceptions.ServiceUnavailable: unavailable: Cache key does not match
ERROR    django.request:log.py:222 Internal Server Error: /health/

---------- coverage: platform darwin, python 3.8.6-final-0 -----------
Name                                         Stmts   Miss Branch BrPart  Cover   Missing
----------------------------------------------------------------------------------------
server/apps/main/models.py                      15      1      2      0    94%   33
server/apps/main/templates/main/index.html      46     46      0      0     0%   1-46
server/apps/main/views.py                        4      1      0      0    75%   12
server/settings/environments/__init__.py         1      1      0      0     0%   2
server/settings/environments/production.py      17     17      0      0     0%   8-75
server/wsgi.py                                   4      4      0      0     0%   10-15
tests/conftest.py                               17      1      2      0    95%   44
----------------------------------------------------------------------------------------
TOTAL                                          243     71      6      0    71%

20 files skipped due to complete coverage.
Coverage HTML written to dir htmlcov

FAIL Required test coverage of 100% not reached. Total coverage: 71.49%
============================== short test summary info ===============================
FAILED tests/test_server/test_urls.py::test_health_check - assert 500 == 200
============================ 1 failed, 6 passed in 2.97s =============================
```

**Comment by codingjoe** [comment]
If it's version dependent, it's probably because now all cache backends get testsed and you have a dummy backend setup ;)
https://github.com/KristianOellegaard/django-health-check/commit/65db560940231b7f104a2cf8a11b88002c8b7a34
This is the commit that breaks your test, and here is your dummy cache https://github.com/wemake-services/wemake-django-template/blob/7f4e9f86882b02e9a8e0c83f7b64b124ad047b61/%7B%7Bcookiecutter.project_name%7D%7D/server/settings/components/caches.py#L8-L10

**Comment by sobolevn** [comment]
Oh, I see! Thanks a lot!
